ports/models.py

```diff
@@ -28,6 +33,236 @@ class Port(models.Model):
+                except KeyError:
+                    pass
+
+                try:
+                    for maintainer in port['maintainers']:
+                        name = maintainer.get('email', {}).get('name', '')
+                        domain = maintainer.get('email', {}).get('domain', '')
+                        github = maintainer.get('github', '')
+
+                        maintainer_object, created = Maintainer.objects.get_or_create(
+                            name=name,
+                            domain=domain,
+                            github=github
+                        )
+
+                        maintainer_object.ports.add(new_port)
+                except KeyError:
+                    pass
+
+                try:
+                    for variant in port['variants']:
+                        variant_object = Variant()
+                        variant_object.port = new_port
+                        variant_object.variant = variant
+                        variant_object.save()
+                except KeyError:
+                    pass
+
+        def load_dependencies_table(ports):
+
+            def load_depends(list_of_dependencies, type_of_dependency, port):
+                dependency = Dependency()
+                dependencies = []
+                dependency.type = type_of_dependency
+                dependency.port_name = port
+                for depends in list_of_dependencies:
+                    try:
+                        dependencies.append(Port.objects.get(name__iexact=depends.rsplit(':', 1)[-1]))
+                    except Port.DoesNotExist:
+                        print("Failed to append {} as a dependency to {}. Not Found.".format(depends.rsplit(':', 1)[-1],
+                                                                                             port.name))
+                dependency.save()
+                dependency.dependencies.add(*dependencies)
+
+            for port in ports:
+                try:
+                    port_object = Port.objects.get(name__iexact=port['name'])
+                    for dependency_type in ["lib", "extract", "run", "patch", "build", "test", "fetch"]:
+                        key = "depends_" + dependency_type
+                        if key in port:
+                            load_depends(port[key], dependency_type, port_object)
+
+                except Port.DoesNotExist:
+                    print("Failed to update dependencies for {}. Port not found in database.".format(port['name']))
+        if isinstance(data, list):
+            load_ports_and_maintainers_table(data)
+            load_categories_table(data)
+            load_dependencies_table(data)
+        elif isinstance(data, str):
+            ports = open_portindex_json(data)
+            load_categories_table(ports)
+            load_ports_and_maintainers_table(ports)
+            load_dependencies_table(ports)
+
+    @classmethod
+    def update(cls, data, is_json=True):
```

Comment:
It seems to me that we are anyway supporting the full `load` functionality in `update`. Do you think we can write `update` slightly more cleverly and remove the need for `load` completely?
cc @mojca

Reply:
`load` is faster than `update` because it does not have to care of any previously available data or relations and thus less number of queries. I had tried to reduce the number of queries for `update` but I believe they should remain more than `load` in any case.

The difference would be only a few minutes on a local database. But with some remote database it might become significant.

Reply:
Carried to #24